### DISCUSSION TITLE
PostgreSQL malformed query when using query builder, joining a sub-query + ManyToMany where-condition + setting a limit

### DISCUSSION THREAD

**Discussion by frw** [mikro-orm/mikro-orm#5538] [General]
### Describe the bug

With the following entities:
```
@Entity()
class User {
  @PrimaryKey()
  id!: number;

  @Property()
  name: string;

  @Property({ unique: true })
  email: string;

  @ManyToMany({ entity: () => Tag, inversedBy: (tag) => tag.users })
  tags = new Collection<Tag, this>(this);

  constructor(name: string, email: string) {
    this.name = name;
    this.email = email;
  }
}

@Entity()
class Tag {
  @PrimaryKey()
  id!: number;

  @Property()
  name: string;

  @ManyToMany({
    entity: () => User,
    mappedBy: (user) => user.tags,
  })
  users = new Collection<User, this>(this);

  constructor(name: string) {
    this.name = name;
  }
}
```

Using a query builder that has a join with a sub-query, a where-condition on a Many-To-Many relationship, and setting a limit results in a malformed query.

Sample code:
```
const result = await orm.em
    .createQueryBuilder(User, "user1")
    .leftJoin(orm.em.createQueryBuilder(User), "user2", {
      "user1.id": orm.em.getKnex().ref("user2.id"),
    })
    .where({ tags: { name: "Tag" } })
    .limit(1)
    .getResult();
```

Resulting query:
```
select "user1".* from "user" as "user1"
where "user1"."id" in (
	select "user1"."id" from (
		select "user1"."id" from "user" as "user1"
		left join (select "u0".* from "user" as "u0") as "user2" // missing join condition
		left join "user_tags" as "u2" on "user1"."id" = "u2"."user_id"
		left join "public"."tag" as "u1" on "u2"."tag_id" = "u1"."id"
		where "u1"."name" = 'Tag'
		group by "user1"."id"
        limit 1
	) as "user1"
)
```

Whereas running the same query without the limit succeeds.

Sample code:
```
const result = await orm.em
    .createQueryBuilder(User, "user1")
    .leftJoin(orm.em.createQueryBuilder(User), "user2", {
      "user1.id": orm.em.getKnex().ref("user2.id"),
    })
    .where({ tags: { name: "Tag" } })
    .getResult();
```

Resulting query:
```
select "user1".* from "user" as "user1"
left join (select "u0".* from "user" as "u0") as "user2" on "user1"."id" = "user2"."id"
left join "user_tags" as "u2" on "user1"."id" = "u2"."user_id"
left join "public"."tag" as "u1" on "u2"."tag_id" = "u1"."id"
where "u1"."name" = 'Tag'
```

### Reproduction

https://github.com/frw/reproduction

The example without the limit succeeds, but the example with the limit fails as the join condition (`on "user1"."id" = "user2"."id"`) is omitted, resulting in a malformed query.

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.2.3-dev.5

### Node.js version

v20.12.2

### Operating system

Ubuntu

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
This should help: 0b851e50b1fa64c76ead91c046aa5d3e231cfab4

Also if the pagination mechanism yields something weird, you can disable that via `qb.setFlag(QueryFlag.DISABLE_PAGINATE)` (but since you are joining a to-many relation here, I guess you want it in place).
